source/plugin/modularization.rst

```diff
@@ -0,0 +1,62 @@
+===================
+Code Modularization
+===================
+
+When writing Sponge plugins, you may find that you need to modularize your code. Maybe you want to split your core into
+another project so that others can make use of it, or maybe you want to utilize a service another plugin provides. With
+Sponge, gone are the days of shading or shadowing entire plugins into your plugin.
+
+Problem 1: Providing Good Service
+=================================
+
+Let's say you want to provide some awesome service for ``WarpService``. You took a day or two to write
+``AwesomeWarpService`` that uses SQL to back warps. So now you can override some other plugin's ``CoreWarpService`` that
+uses mere flatfiles with your own service.
+
+But when you go to use ``Sponge.getServiceManager().provide(WarpService.class)``, you are shocked to find that referencing
+``WarpService`` throws a ``ClassNotFoundException``!
+
+What's Happening
+~~~~~~~~~~~~~~~~
+
+When you compile your plugin, the compiler finds ``WarpService.class`` because it's present in the workspace (either via
+a build tool like Gradle or by just adding it to the classpath). However, when Sponge goes to run your plugin, it can't
+find the referenced ``WarpService.class`` in your plugin's "container".
+
+Solution
+~~~~~~~~
+
+Using Sponge's dependency system, you can declare hard and soft dependencies using the ``dependencies`` field of the
+``@Plugin`` tag. Here's an example:
+
+.. code-block:: java
+
+    @Plugin(
+      id = "Example",
+      name = "Example",
+      version = "1.0-SNAPSHOT",
+      description = "Example Plugin",
+      url = "https://example.com",
+      authors = { "AniSkywalker" },
```

Comment:
Personally I'd leave my own name out of the docs in place of something such as 'Spongie'.